qCC/ccGraphicalSegmentationTool.cpp

```diff
@@ -49,1600 +50,1616 @@
+                    CCVector3d Pd(half_w + Pscreen->x, half_h + Pscreen->y, 0/*Pscreen->z*/);
+                    CCVector3d Q3D;
+                    camera.unproject(Pd, Q3D);
+                    *Pscreen = Q3D.toPC();
+                }
+                verticesPC->invalidateBoundingBox();
+            }
+            else
+            {
+                assert(false);
+                ccLog::Warning("[Segmentation] Failed to convert 2D polyline to 3D! (internal inconsistency)");
+                mode2D = false;
+            }
+
+            //export Global Shift & Scale info (if any)
+            bool hasGlobalShift = false;
+            CCVector3d globalShift(0, 0, 0);
+            double globalScale = 1.0;
+            {
+                for (QSet<ccHObject *>::const_iterator it = m_toSegment.constBegin(); it != m_toSegment.constEnd(); ++it)
+                {
+                    ccShiftedObject *shifted = ccHObjectCaster::ToShifted(*it);
+                    bool isShifted = (shifted && shifted->isShifted());
+                    if (isShifted)
+                    {
+                        globalShift = shifted->getGlobalShift();
+                        globalScale = shifted->getGlobalScale();
+                        hasGlobalShift = true;
+                        break;
+                    }
+                }
+            }
+
+            if (hasGlobalShift && m_toSegment.size() != 1)
+            {
+                hasGlobalShift = (QMessageBox::question(MainWindow::TheInstance(), "Apply Global Shift", "At least one of the segmented entity has been shifted. Apply the same shift to the polyline?", QMessageBox::Yes, QMessageBox::No) == QMessageBox::Yes);
+            }
+
+            if (hasGlobalShift)
+            {
+                poly->setGlobalShift(globalShift);
+                poly->setGlobalScale(globalScale);
+            }
+        }
+
+        QString polyName = QString("Segmentation polyline #%1").arg(++s_polylineExportCount);
+        poly->setName(polyName);
+        poly->setEnabled(false); //we don't want it to appear while the segmentation mode is enabled! (anyway it's 2D only...)
+        poly->set2DMode(mode2D);
+        poly->setColor(ccColor::yellow); //we use a different color so as to differentiate them from the active polyline!
+
+        //save associated viewport
+        cc2DViewportObject *viewportObject = new cc2DViewportObject(polyName + QString(" viewport"));
+        viewportObject->setParameters(m_associatedWin->getViewportParameters());
+        viewportObject->setDisplay(m_associatedWin);
+        poly->addChild(viewportObject);
+
+        mainWindow->addToDB(poly, false, false, false);
+        ccLog::Print(QString("[Segmentation] Polyline exported (%1 vertices)").arg(poly->size()));
+    }
+}
+
+void ccGraphicalSegmentationTool::options()
+{
+    ccGraphicalSegmentationOptionsDlg optionsDlg("Segmentation Options", this);
+
+    if (!optionsDlg.exec())
+        return;
 }
 
```

Comment:
remove white spaces

Reply:
Have you removed these 2 blank lines?

Reply:
First one is the empty one on 1292. Is there a second one?